doc/macsec/MACsec_hld.md

```diff
@@ -0,0 +1,743 @@
+  - [4.3 Create Ingress/Egress SC](#43-create-ingressegress-sc)
+  - [4.4 Create and Enable Ingress SA](#44-create-and-enable-ingress-sa)
+  - [4.5 Create and Enable Egress SA](#45-create-and-enable-egress-sa)
+  - [4.6 Disable and Remove Egress SA](#46-disable-and-remove-egress-sa)
+  - [4.7 Disable and Remove Ingress SA](#47-disable-and-remove-ingress-sa)
+  - [4.8 Remove Ingress/Egress SC](#48-remove-ingressegress-sc)
+  - [4.9 MACsec Deinit](#49-macsec-deinit)
+  - [4.10 Deinit Port](#410-deinit-port)
+
+## About this Manual
+
+This document provides general information about the MACsec feature implementation in SONiC.
+
+## Abbreviation
+
+| Abbreviation | Description                                  |
+| ------------ | -------------------------------------------- |
+| CA           | Secure Connectivity Association              |
+| CAK          | Secure Connectivity Association Key          |
+| CKN          | Secure Connectivity Association Key Name     |
+| EAPOL        | Extensible Authentication Protocol over LANs |
+| KaY          | MAC Security Key Agreement Entity            |
+| MKA          | MACsec Key Agreement protocol                |
+| SA           | Secure Association                           |
+| SAK          | Secure Association Key                       |
+| SC           | Secure Channel                               |
+| SCI          | Secure Channel Identifier                    |
+| XPN          | Extension Packet Number                      |
+
+## 1 Requirements Overview
+
+### 1.1 Functional requirements
+
+This section describes the SONiC requirements for MACsec primary feature.
+
+At a high level the following should be supported:
+
+#### Phase I
+
+- MACsec can be enable at a specified [port](https://github.com/Azure/SONiC/wiki/Configuration#port)
+- MACsec can co-work with the [port channel](https://github.com/Azure/SONiC/wiki/Configuration#port-channel)
+- Support Cipher: GCM-AES-128 and GCM-AES-256
+- Secure Association Key(SAK) can be replaced without service outage
+
+#### Phase II
+
+- MACsec can support Extension packet number(XPN), which means to support Cipher Suites: GCM-AES-XPN-128 and GCM-AES-XPN-256
+- SAK can be refreshed proactively.
+- Primary and Fallback secure Connectivity Association Key can be supported simultaneously.
+- Enable or disable the XPN feature by the wpa_cli
+- Parameters of wpa_supplicant, send_sci, replay_protect, replay_window_size and rekey_period, can be updated on the fly
+- CLI command `show macsec` to monitor mka session and statistics of MACsec
+
+#### Phase III
+
+- CLI commands to configure MACsec
+
+#### Phase IV
+
+- MACsec can support confidentiality offset setting
+
+## 2 Architecture Design
+
+This chapter shows the MACsec interface stack of virtual switch and real switch.
+
+Virtual switch use the Linux MACsec driver as the MACsec Security Entity(SecY) to support the functionality of MACsec and the SecY is imposed on the physical port.
```

Comment:
>Virtual switch [](start = 0, length = 14)

I know you are talking about SONiC. But it is easy to mislead to the virtual switch between VMs  on a physical server. And in SONiC terminology, we already have a 'Virtual Switch SAI', I think your term means total different thing.

Reply:
> > Virtual switch [](start = 0, length = 14)
> 
> I know you are talking about SONiC. But it is easy to mislead to the virtual switch between VMs on a physical server. And in SONiC terminology, we already have a 'Virtual Switch SAI', I think your term means total different thing.

I discussed with Guohan and change it to SAI virtual switch, please review whether it makes sense.